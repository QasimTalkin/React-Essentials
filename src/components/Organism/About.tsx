import React, { useState } from "react";
import PokeGallery from "../Molecule/PokeGallery";
const DEMO_MODE = false;
export default function About() {
  let seconds = 0;

  function startStopwatch() {
    setInterval(() => {
      console.log(seconds);
      seconds += 1;
    }, 1000);
  }
  console.log(process.env);
  const [count, setCount] = useState(0);
  let [limit, setLimit] = useState(7);
  
  
  return (
    // give section some padding and background color
    <section className="bg-emerald-20 px-10">
      {DEMO_MODE && (
        <>
          <div>
            {seconds}
            <button className="bg-red-500" onClick={startStopwatch}>
              Start
            </button>
          </div>
          <div>
            <p>You clicked {count} times</p>
            <button onClick={() => setCount(count + 1)}>Click me</button>
          </div>
        </>
      )}

      <h1 className="text-3xl font-bold underline text-center">Welcome </h1>
      {/* large input box to set the limit */}
      <div className="grid gap-2 p-10">
        <input type="number" className="w-full border-2 border-gray-300 p-2" onChange={(e) => setLimit(parseInt(e.target.value))}/>
      </div>
      <PokeGallery limit={limit}/>
      {/* <img
        src="https://images.unsplash.com/photo-1561336313-0bd5e0b27ec8?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=3540&q=80"
        style={{ width: "100%" }}
        alt="cover pic"
        id="cover-pic"
      /> */}
    </section>
  );
}
