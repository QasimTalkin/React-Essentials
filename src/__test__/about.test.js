import React from 'react';
import { render, cleanup } from '@testing-library/react'; 
import About from '../components/Organism/About'

afterEach(cleanup);

describe('About Component renders correctly', () => {

  it ('renders correctly', () => {
    const { asFragment } = render(<About />);
    expect(asFragment()).toMatchSnapshot();
  });
  
  test('rendered image contains alt text', ()=> {
  
    const { getByAltText } = render(<About />);
    const image = getByAltText(/pic/i);
    expect(image).toBeInTheDocument();
  });
  
});