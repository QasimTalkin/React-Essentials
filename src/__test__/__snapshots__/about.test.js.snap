// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Component renders correctly renders correctly 1`] = `
<DocumentFragment>
  <section>
    <h1
      class="text-3xl font-bold underline"
    >
      Welcome Abul
    </h1>
    <img
      alt="cover pic"
      id="cover-pic"
      src="https://images.unsplash.com/photo-1561336313-0bd5e0b27ec8?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=3540&q=80"
      style="width: 100%;"
    />
  </section>
</DocumentFragment>
`;
